mlflow/utils/autologging/__init__.py

```diff
@@ -954,7 +955,10 @@ def patch_with_managed_run(original, *args, **kwargs):
 
             try:
                 result = patch_function(original, *args, **kwargs)
-            except:
+            except (Exception, KeyboardInterrupt):
+                # In addition to standard Python exceptions, handle keyboard interrupts to ensure
+                # that runs are terminated if a user prematurely interrupts training execution
+                # (e.g. via sigint / ctrl-c)
                 if managed_run:
                     try_mlflow_log(mlflow.end_run, RunStatus.to_string(RunStatus.FAILED))
```

Comment:
Ending the run with status `FAILED` in response to a keyboard interrupt is consistent with the behavior of `with mlflow.start_run()` (see https://github.com/mlflow/mlflow/blob/b57f95498ac11d70d3d153ce5e0895a7ab4ab7e9/mlflow/tracking/fluent.py#L100).

While `KILLED` may make more sense, I'd rather maintain consistency and avoid making backwards-incompatible changes.